### PR TITLE
test/e2e/framework: remove direct dependency to k8s.io/kubernetes/pkg/controller

### PR DESCRIPTION
**Pull request by tanjunchen**
**What type of PR is this?**
  /kind cleanup
 

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

ref:https://github.com/kubernetes/kubernetes/issues/74352
**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

### CODE REVIEW

**Comment by tanjunchen** [comment]
/priority important-longterm
/kind cleanup

/cc @neolit123 @oomichi
Thank you very much.

File: test/e2e/framework/resource/resources.go [hunk 1]
```diff
@@ -175,6 +175,26 @@ func waitForPodsInactive(ps *testutils.PodStore, interval, timeout time.Duration
 	return err
 }
```
Comment (tanjunchen) [inline test/e2e/framework/resource/resources.go]: k8s.io/kubernetes/pkg/controller source code.

```
// FilterActivePods returns pods that have not terminated.
func FilterActivePods(pods []*v1.Pod) []*v1.Pod {
	var result []*v1.Pod
	for _, p := range pods {
		if IsPodActive(p) {
			result = append(result, p)
		} else {
			klog.V(4).Infof("Ignoring inactive pod %v/%v in state %v, deletion time %v",
				p.Namespace, p.Name, p.Status.Phase, p.DeletionTimestamp)
		}
	}
	return result
}

func IsPodActive(p *v1.Pod) bool {
	return v1.PodSucceeded != p.Status.Phase &&
		v1.PodFailed != p.Status.Phase &&
		p.DeletionTimestamp == nil
}

```

**Comment by alejandrox1** [comment]
:thinking: CI...
/retest

File: test/e2e/framework/resource/resources.go [hunk 2]
```diff
@@ -175,6 +175,26 @@ func waitForPodsInactive(ps *testutils.PodStore, interval, timeout time.Duration
...
 
+// filterActivePods returns pods that have not terminated.
+func filterActivePods(pods []*v1.Pod) []*v1.Pod {
+	isPodActive := func(p *v1.Pod) bool {
+		return v1.PodSucceeded != p.Status.Phase &&
+			v1.PodFailed != p.Status.Phase &&
+			p.DeletionTimestamp == nil
+	}
```
Comment (alejandrox1) [inline test/e2e/framework/resource/resources.go]: Let's leave this one as a separate function.
It is handy and it is used in some other e2e tests.
So we can get this framework onto staging and add another utility function to the e2epod pkg.

File: test/e2e/framework/resource/resources.go [hunk 3]
```diff
@@ -175,6 +175,26 @@ func waitForPodsInactive(ps *testutils.PodStore, interval, timeout time.Duration
 	return err
 }
 
+// filterActivePods returns pods that have not terminated.
+func filterActivePods(pods []*v1.Pod) []*v1.Pod {
```
Comment (alejandrox1) [inline test/e2e/framework/resource/resources.go]: This function seems to only be used (in k/k/test) here.
Though given the name and the fact that it uses `IsPodActive()` (which would definitely fit in into the e2epod pkg), I wonder whether we should keep it here, e2eresource, or move it over to e2epod.
I vote for e2epod just based on the sort of logic.

**Review by alejandrox1 — CHANGES_REQUESTED** [review]
Thank you @tanjunchen

File: test/e2e/framework/resource/resources.go (hunk 3 above)
Comment (tanjunchen) [reply test/e2e/framework/resource/resources.go]: +1  for e2epod 😄

File: test/e2e/framework/resource/resources.go (hunk 2 above)
Comment (tanjunchen) [reply test/e2e/framework/resource/resources.go]: sure

**Comment by tanjunchen** [comment]
/cc @alejandrox1 
updated

**Review by SataQiu — COMMENTED** [review]
Nice work!
/lgtm

**Pull request merged**